City-of-Helsinki/drupal-helfi-platform-config
# Draft: UHF-4792: Add update hook for TPR view changes for new status_extra filter

# Config for status_extra filter for TPR entities [UHF-4792](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4792)
Takes new custom views filter into use on TPR entity views.

## What was done
- Moves TPR views from helfi_tpr module to helfi_tpr_config
- Updates end user facing views with new filter, which displays entities if they are published OR the user has permission to view unpublished entities

## How to install
* Make sure your instance (for example, SOTE) is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Login to the site and open the following views so you can compare the changes and see if any filters / fields are missing after the updates:
  * /admin/structure/views/view/unit_services
  * /admin/structure/views/view/service_units
  * /admin/structure/views/view/service_list
  * admin/structure/views/view/unit_search
* Update the Helfi TPR module to the latest release:
  * `composer require "drupal/helfi_tpr:^2.0.5"`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-4792-tpr-views-config`
* Run `make drush-updb drush-cr`

## How to test
* Check the previously opened views:
  * The default / block displays should have the new "TPR - [Entity name]: Julkaisutila tai ylläpitokäyttäjä." filter enabled and the the "TPR - [Entity name]: Käännöksen tila" should be missing.
  * For the entity reference displays the "Käännöksen tila" filter should still be present and not replaced with the new filter.
  * See if any fields or filters are missing or any configuration options got broken
